src/IO/H5/File.cpp

```diff
@@ -88,8 +91,21 @@ H5File<Access_t>& H5File<Access_t>::operator=(H5File&& rhs) {
 template <AccessType Access_t>
 H5File<Access_t>::~H5File() {
   if (file_id_ != -1) {
-    CHECK_H5(H5Fclose(file_id_),
-             "Failed to close file: '" << file_name_ << "'");
+    const auto h5_status = H5Fclose(file_id_);
+    if (h5_status < 0) {
```

Comment:
While squashing, please add a comment explaining briefly what's going on here in the destructor, and why you made this design choice.